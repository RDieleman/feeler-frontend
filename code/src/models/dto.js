export function AddBookDTO(bookshelfId, isbn){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
}

export function GetBookshelfDTO(bookshelfId){
    this.bookshelfId = bookshelfId;
}

export function RemoveBookDTO(bookshelfId, isbn){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
}

export function UpdateBookDTO(bookshelfId, isbn, status){
    this.bookshelfId = bookshelfId;
    this.isbn13 = isbn;
    this.status = status;
}

