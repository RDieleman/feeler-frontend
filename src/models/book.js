//Book info
export function Book(title, overview, isbn, imageUrl){
    this.title = title;
    this.overview = (overview) ? overview : "No description found.";
    this.isbn = isbn;
    this.imageUrl = imageUrl;
}