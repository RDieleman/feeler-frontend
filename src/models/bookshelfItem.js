export function BookshelfItem(id, isbn, status){
    this.id = id;
    this.isbn = isbn;
    this.status = status;
}