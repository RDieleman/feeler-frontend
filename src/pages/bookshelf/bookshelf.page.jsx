import React, {Component} from "react";
import "./bookshelf.styles.css";
import {PaddingComponent} from "../../components/layout/padding/padding.component";
import {properties} from "../../properties";
import {
    ButtonMainNumberComponent
} from "../../components/input/button/button-main/button-main-number/button-main-number.component";
import {ButtonSecNumberComponent} from "../../components/input/button/button-sec/button-sec-number/button-sec-number.component";

class BookshelfPage extends Component {

    constructor(props, context) {
        super(props, context);
        this.state = {};
    };

    //redirect to finished books
    handleFinishedClick = () => {
        this.props.history.push("/shelf/read");
    }

    //Redirect to currently reading
    handleReadingClick = () => {
        this.props.history.push("/shelf/reading");
    }

    //Redirect to unread
    handleUnreadClick = () => {
        this.props.history.push("/shelf/unread");
    }

    render() {
        const {bookshelf} = this.props;

        const read = [];
        const reading = [];
        const unread = [];

        bookshelf.content.forEach(i => {
            switch (i.status) {
                case "READ":
                    read.push(i);
                    return;
                case "READING":
                    reading.push(i);
                    return;
                case "UNREAD":
                default:
                    unread.push(i);
                    break;
            }
        })

        return (
            <div id="page-container">
                <div id="shelf-container" className="container-horizontal">

                    <PaddingComponent/>
                    <div id="shelf-content" className="container-vertical">

                        <PaddingComponent/>
                        <PaddingComponent/>

                        {/*Main text*/}
                        <div id="shelf-text-container" className="text-body container-vertical">
                            {properties.textShelfGreeting}
                        </div>

                        {/*User options*/}
                        <div className="container-vertical">

                            <div className="text-body">{
                                properties.textShelfAnswer}
                            </div>

                            {/*Button with redirect to read page*/}
                            <ButtonMainNumberComponent
                                handleOnClick={this.handleFinishedClick}
                                content={properties.textShelfBtnRead}
                                number={read.length}
                            />

                            <PaddingComponent basis="10px"/>

                            {/*Button with redirect to reading page*/}
                            <ButtonMainNumberComponent
                                handleOnClick={this.handleReadingClick}
                                content={properties.textShelfBtnReading}
                                number={reading.length}
                            />

                            <PaddingComponent/>

                            {/*Button with redirect to unread page*/}
                            <ButtonSecNumberComponent
                                handleOnClick={this.handleUnreadClick}
                                content={properties.textShelfBtnUnread}
                                number={unread.length}
                            />
                        </div>
                    </div>
                    <PaddingComponent/>
                </div>

                <PaddingComponent/>
            </div>
        )
    }
}

export default BookshelfPage;