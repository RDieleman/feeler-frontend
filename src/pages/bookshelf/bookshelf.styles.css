#shelf-container{
    flex: 1;
}

#shelf-text-container{
    flex: 1;
}