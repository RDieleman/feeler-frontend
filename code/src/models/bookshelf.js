//Bookshelf
import {BookshelfItem} from "./bookshelfItem";

export function Bookshelf(id, content){
    this.id = id;
    this.content = [];
    content.forEach(i => {
        this.content.push(
            new BookshelfItem(
                i.id,
                i.isbn13,
                i.status
            )
        );
    })
    this.getRead = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "READ") {
                items.push(i);
            }
        });
        return items;
    }

    this.getUnread = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "UNREAD") {
                items.push(i);
            }
        });
        return items;
    }

    this.getReading = () =>{
        const items = [];
        this.content.forEach(i => {
            if (i.status === "READING") {
                items.push(i);
            }
        });
        return items;
    }
}