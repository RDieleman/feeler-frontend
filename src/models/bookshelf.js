//Bookshelf
import {BookshelfItem} from "./bookshelfItem";

export function Bookshelf(id, content){
    this.id = id;
    this.content = [];
    content.forEach(i => {
        this.content.push(
            new BookshelfItem(
                i.id,
                i.isbn,
                i.status
            )
        );
    })
}