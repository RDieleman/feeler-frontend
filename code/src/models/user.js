import {Bookshelf} from "./bookshelf";

export function User(id, name, bookshelf){
    this.id = id;
    this.name = name;
    this.bookshelf = new Bookshelf(
        bookshelf.id,
        bookshelf.content
    );
}