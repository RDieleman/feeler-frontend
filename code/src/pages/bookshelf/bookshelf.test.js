import React from "react";
import {shallow} from "enzyme";
import BookshelfPage from "./bookshelf.page";
import {Bookshelf} from "../../models/bookshelf";
import {Book} from "../../models/book";

it("Expect to render Bookshelf page", () => {
    const books = [];

    for(let i; i < 5; i++){
        books.push(new Book(
            "test title" + i,
            "test overview" + i,
            "test isbn" + i,
            "test image url" + i
        ))
    }

    const bookshelf = new Bookshelf(1, books)

    expect(shallow(
        <BookshelfPage
            bookshelf={bookshelf}
        />)).toMatchSnapshot();
});