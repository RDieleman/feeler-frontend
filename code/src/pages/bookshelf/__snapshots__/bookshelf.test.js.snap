// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect to render Bookshelf page 1`] = `ShallowWrapper {}`;
